source/lac/sparsity_tools.cc

```diff
@@ -39,15 +39,270 @@ extern "C"
 DEAL_II_ENABLE_EXTRA_DIAGNOSTICS
 #endif
 
+#include <zoltan_cpp.h>
```

Comment:
we need to be more careful here as `Trilinos` might not be compiled with `Zoltan` package, please add `Zoltan` to 
this line https://github.com/dealii/dealii/blob/master/cmake/configure/configure_2_trilinos.cmake#L58 and guard these includes and code in
```
#ifdef DEAL_II_TRILINOS_WITH_Zoltan
...
#endif
```

Reply:
it would probably be better to use `STRING(TOUPPER "${_lib}" _lib_upper)`, something like

```
    FOREACH(_optional_module
   	  ROL Zoltan
      )
      ITEM_MATCHES(_module_found ${_optional_module} ${Trilinos_PACKAGE_LIST})
      IF(_module_found)
        MESSAGE(STATUS "Found ${_optional_module}")
        STRING(TOUPPER "${_optional_module}" _optional_module_upper)
        SET(DEAL_II_TRILINOS_WITH_${_optional_module_upper} ON)
      ELSE()
        MESSAGE(STATUS "Module ${_optional_module} not found!")
      ENDIF()
    ENDFOREACH()
```
so that we can write
```
DEAL_II_TRILINOS_WITH_ZOLTAN
```
consistent with all other macros.

Reply:
Also guard the include with DEAL_II_DISABLE_EXTRA_DIAGNOSTICS